aws/main.yaml

```diff
@@ -12,10 +12,15 @@ Parameters:
     Description: >-
       API key for the Datadog account (find at
       https://app.datadoghq.com/account/settings#api)
+      It will be stored in AWS Secrets Manager securely. If DdApiKeySecretArn is also set, this value is ignored.
     Type: String
     NoEcho: true
-    AllowedPattern: .+
-    ConstraintDescription: DdApiKey is required
+    Default: ""
+  DdApiKeySecretArn:
+    Type: String
+    AllowedPattern: "arn:.*:secretsmanager:.*"
+    Default: "arn:aws:secretsmanager:DEFAULT"
```

Comment:
I'd like to not have a default value for this parameter. The default value will interact weirdly with users usually filling in the api key manually - users will then have to delete this field.

Reply:
I could delete the default parameter and change the conditions. Although right now, the user doesn't need to remove this field when they set the api key manually. Since the value would be piped to the forwarder template and if it finds that default value, it will use the api key, not the Secrets Manager ARN.

Refer to https://github.com/DataDog/datadog-serverless-functions/blob/1c3523b74a9549c00c2fc112591bf56c44f1f553/aws/logs_monitoring/template.yaml#L647-L654
Let me know what you think about this. 

I'm not sure what happens if you don't send a parameter to a template, for example, the forwarder template has a default value of `arn:aws:secretsmanager:DEFAULT` for the Secrets Manager ARN. Even if we don't set a default value here to be piped, would it end up being the same default in the forwarder? Or will it be sent as empty?

Reply:
Ah okay that seems really weird from a user perspective. I don't think this is particularly intuitive. Instead can we have the Default in the parameter be `""`, change the condition appropriately, and then when we pass it as a parameter to the forwarder we can change it to this "arn:aws:secretsmanager:DEFAULT" string. Here's documentation on [cf if statements](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-conditions.html#intrinsic-function-reference-conditions-if), and [conditions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/conditions-section-structure.html).

Reply:
Updated and added requested changes.